lib/SILGen/SILGen.cpp

```diff
@@ -763,6 +763,31 @@ void SILGenModule::emitAbstractFuncDecl(AbstractFunctionDecl *AFD) {
           SILReverseDifferentiableAttr::create(
             M, indices, primName, adjName,
             /*primitive*/ hasPrimitiveAdjoint));
+
+      // Register the JVP and VJP as "emitted" so that the witness table emitter
+      // finds them.
+      // TODO: We shoulde emit blank JVP and VJP functions in SILGen when
+      // they're not user-specified, so that the witness table can also have
+      // with non-user-specified functions.
+      auto originalDeclRef = SILDeclRef(AFD);
+      if (auto *jvpFnDecl = diffAttr->getJVPFunction()) {
+        auto declRef = originalDeclRef.asAutoDiffAssociatedFunction(
+            AutoDiffAssociatedFunctionIdentifier::get(
+                AutoDiffAssociatedFunctionKind::JVP,
+                diffAttr->getCheckedParameterIndices(),
+                AFD->getASTContext()));
+        emittedFunctions[declRef] = getFunction(SILDeclRef(jvpFnDecl),
```

Comment:
Maybe use insert() instead of assigning to the result of operator[] and assert that it was inserted?